remove pyzmail36 dependency and add mail-parser

pmendezsuarez opened the pull request:
No new features introduced.
Removed pyzmail36 dependency to parse e-mails and replaced for mail-parser
Tests passed.
Also changed Django version (>4.0) in 'demo' to avoid url() function problems.

sastred commented:
Hi @pmendezsuarez !

Thanks for your PR. All changes look good to me :+1: 

Unfortunately, it's not enough. There are still some references to `pyzmail36` in `admin.py` and `models.py` that need to be replaced. I recommend you to do a global search for `pyz` in all the repo. Here is an example from @aaloy's PR that it's not replaced in your PR: https://github.com/APSL/django-yubin/pull/59/files?authenticity_token=q6QxQCUyycDXjGsRzECaP9NoG3QlreEr6Mcx0MlzQied9%2Fxhwlwysson9b4xdNBL0f6qBv2LLUE1Mj%2BdSZLSSw%3D%3D&file-filters%5B%5D=.py#diff-acc4d171dd3889243f121103e3b7c443d7f7c5bc959ec7e4d7a1dae582ba7d9bL73

From a user point of view, I have found the error trying to view an email in `/admin/django_yubin/message/` clicking in the `show` link. If you didn't get an error probably is because you should have `pyzmail36` installed in your virtualenv. Try to uninstall it so you can also get the errors.

![image](https://user-images.githubusercontent.com/2037027/147107221-5c91a251-4a80-4918-8d15-fc87578d8ff0.png)

pmendezsuarez commented:
Hi @sastred

Thanks for your response. Thanks to your observations and using @aaloy's PR as a guide, I implemented a few changes.
Now all _pyzmail36_ references are replaced and the functions that were making use of this library (mostly in _admin.py_) are now refactored with _mail-parser_ functionality.

About the error you got in _/admin/django_yubin/message/_ is now fixed. It was related to the _html_view_ method, that was using the _pyzmail_ dependency, now is refactored and is working again.
